[ADD] milestone 메인페이지

marulloc opened the pull request:
[comment]: <> (PR 태그를 명시해주세요. [ADD] / [UPDATE] / [BUG])

### 요약
요구사항에 정의된 milestone 메인페이지 추가

### 핵심 로직 및 내용
milestone과 연관된 issue를 group by로 가져온 후, 데이터 후처리를 통해 클라이언트에 전송
클라이언트는 렌더링

### 기타 참고 사항
sql문만 보시면 됩니다. 그외의 부분은 다른 컴포넌트와 로직 비슷합니다

marulloc committed:
[FEAT] ADD PAGE, API

[why]
basic setting

[what]
add API & route logic
add page rendering method

[etc]
etc...

marulloc committed:
[FEAT] MILESTONE MAIN PAGE

[why]
show milestone list

[what]
add sql query statement that find issues based on milestone
add milestone.js

[etc]
etc...

marulloc committed:
[FEAT] MILESTONE GET

[why]
for drawing milestone progress bar
and minimize number of request(to db)

[what]
add api /milestone/count
add router logic
add data processing stage

[etc]
etc...

thesulks commented on Backend/Routes/milestoneRoute.js:
```diff
@@ -1,9 +1,48 @@
+  const sql = `
+  SELECT mi.id, mi.isOpen as mileIsOpen, mi.dueDate, mi.description, mi.title, iss.isOpen as issueIsOpen, COUNT(iss.id) as cnt
+  FROM milestones mi LEFT JOIN issues iss ON mi.id = iss.milestoneId
+  GROUP BY mi.id, iss.isOpen ORDER BY mi.id
```
아래 로직대로 새 객체에 넣어준다면 `ORDER BY`가 필요없을 것 같기도 합니다🙃

marulloc replied:
맞습니당

thesulks commented on Backend/Routes/milestoneRoute.js:
```diff
@@ -1,9 +1,48 @@
+  const result = {};
+
+  // eslint-disable-next-line array-callback-return
+  rows.map((row) => {
```
배열을 새로 매핑(`map`)해서 새로운 배열을 만드는 작업이 아닌 것 같은데 `forEach`가 더 적절할 것 같은데 같이 얘기해보면 좋을 것 같습니다

pje1740 replied:
eslint 가 뭔가 취소된 흔적이 있는데 가급적 이렇게 수동으로 취소하지 않고 해결해보면 좋을 거 같습니다~ 도균님 말씀처럼 map의 반환값이 없어서인가 싶기도 하네요! :)

marulloc replied:
백번 맞습니당

pje1740 commented on Frontend/Views/Pages/LabelPage.js:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const LabelPage = () => {
```
이거는 향후에 제 걸로 덮어쓰기 될 수도 있을 거 같습니다! 제가 라벨 페이지 작업 중이라서 PR 할 시점에 다시 한번 언급 드릴게요 :)

marulloc replied:
맘껏 덮어씌우시면 됩니당~

pje1740 commented on Backend/Routes/milestoneRoute.js:
```diff
@@ -1,9 +1,48 @@
-router.get('/', async (req, res, next) => {
-  const sql = `SELECT * FROM milestones`;
+router.get('/count', async (req, res) => {
+  const sql = `
```
사소한 부분 같기는 한데, 변수명을 sql 보다는 query 같은 걸로 적는 것이 의미상 맞지 않나 싶습니다! sql은 언어 자체를 의미하는 거라 맥락이 좀 다를 거 같아요 ㅎㅎ

marulloc replied:
천번 맞습니당

pje1740 commented on Frontend/Views/Components/Milestone.js:
```diff
@@ -0,0 +1,29 @@
+  };
+
+  const box2 = {
+    width: `${milestone.total && (milestone.closedIssue / milestone.total) * 200}px`,
```
200을 static하게 적지 않고 따로 변수 처리를 하면 좋을 거 같습니다!
수식으로 한방에 처리한 거 멋져요!! 👍

marulloc replied:
이건 css작업 들어가면 변경하도록 하겠습니당 ㅋㅋㅋㅋ

pje1740 reviewed (approved):
굿굿!! 오전에 논의한 거처럼 한번의 API 요청으로 데이터 받아올 수 있도록 바꾼 게 다른 페이지들하고도 일관성이 있는 거 같아 좋습니다!!
고생 하셨습니다!!

marulloc committed:
[STYLE] Apply feedback

[why]
beautify...
meaningful variable name

[what]
alter variable name 'sql' to 'query'
alter map() to forEach()
delete Order by in query

[etc]
etc...

pje1740 reviewed (approved):
LGTM!
